test/controllers/csvs_controller_test.rb

```diff
@@ -0,0 +1,335 @@
+require_relative '../test_helper'
+
+describe CsvsController do
+  let(:deployer) { users(:deployer) }
+  let(:export) { csv_exports(:pending) }
+
+  describe "permissions test" do
+    as_a_admin do
+      describe "a GET to new" do
+        it "renders the page with user link & admin menu" do
+          get :new
+          assert_template :new
+          assert @response.body.include? "Create Deploys CSV Report"
+          assert @response.body.include? "Download Users CSV Report"
+          assert @response.body.include? "Admin"
+          assert @response.body.include? "Environment variables"
+        end
+      end
+    end
+
+    as_a_deployer do
+      describe "a GET to new" do
+        it "renders the page without user link & limited admin menu" do
+          get :new
+          assert_template :new
+          assert @response.body.include? "Create Deploys CSV Report"
+          refute @response.body.include? "Download Users CSV Report"
+          assert @response.body.include? "Admin"
+          refute @response.body.include? "Environment variables"
+        end
+      end
+    end
+  end
+
+  as_a_admin do
+    describe "a GET to index for a user with csv exports" do
+      setup { get :index, format: format }
+    
+      describe "as html" do
+        let(:format) { :html }
+      
+        it "renders the template" do
+          assert_template :index
+        end
+      end
+      
+      describe "as JSON" do
+        let(:format) { :json }
+        
+        it "renders json" do
+          assert_equal "application/json", @response.content_type
+          assert_response :ok
+        end
+      end
+    end
+  end
+
+  as_a_viewer do
+    describe "a GET to index for a user with no csv exports" do
```

Comment:
this is changing 2 different things ... ideally a test should only change 1 thing
also there is no special `0 exports` logic ... so this amount of tests would be good enough ...

```Ruby
as_a_viewer do
  describe "#index" do
    it "renders html" do
      get :index, format: :html
      assert_template :index
    end

    it "renders csv" do
      get :index, format: :csv
     @response.content_type.must_equal  "application/json"
    end
  end
end
```